### PR TITLE
[Material] Decoration for DataTable is not used inside PaginatedDataTable

### PR DESCRIPTION
**Pull request by perclasson**
## Description

`PaginatedDataTable` is a `DataTable` wrapped in a `Card`. With this change, the  decoration from the DataTableThemeData is not used inside of it. It could cause having multiple background colors or borders.

## Related Issues

https://github.com/flutter/flutter/issues/69766

## Tests

I added the following tests:

- Table has no decoration with DataTableTheme

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I signed the [CLA].
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] All existing and new tests are passing.
- [X] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [X] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [X] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. 

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Review by guidezpl — APPROVED** [review]
```
  _    ___ _____ __  __ 
 | |  / __|_   _|  \/  |
 | |_| (_ | | | | |\/| |
 |____\___| |_| |_|  |_|
                        
```

File: packages/flutter/test/material/paginated_data_table_test.dart [hunk 1]
```diff
@@ -786,4 +786,34 @@ void main() {
     await tester.pumpWidget(buildTable(false));
     expect(find.byType(Checkbox), findsNothing);
   });
+
+  testWidgets('Table has no decoration with DataTableTheme', (WidgetTester tester) async {
```
Comment (rami-a) [inline packages/flutter/test/material/paginated_data_table_test.dart:790]: Reword this to say something like `Table should not use decoration from DataTableTheme`

File: packages/flutter/lib/src/material/paginated_data_table.dart [hunk 2]
```diff
@@ -471,6 +471,7 @@ class PaginatedDataTableState extends State<PaginatedDataTable> {
                     sortColumnIndex: widget.sortColumnIndex,
                     sortAscending: widget.sortAscending,
                     onSelectAll: widget.onSelectAll,
+                    decoration: const BoxDecoration(),
```
Comment (rami-a) [inline packages/flutter/lib/src/material/paginated_data_table.dart:474]: Can you add a comment for why you do this.

**Review by rami-a — APPROVED** [review]

File: packages/flutter/test/material/paginated_data_table_test.dart (hunk 1 above)
Comment (perclasson) [reply packages/flutter/test/material/paginated_data_table_test.dart:790]: Done!

File: packages/flutter/lib/src/material/paginated_data_table.dart (hunk 2 above)
Comment (perclasson) [reply packages/flutter/lib/src/material/paginated_data_table.dart:474]: Done!

File: packages/flutter/lib/src/material/paginated_data_table.dart [hunk 3]
```diff
@@ -471,6 +471,9 @@ class PaginatedDataTableState extends State<PaginatedDataTable> {
                     sortColumnIndex: widget.sortColumnIndex,
                     sortAscending: widget.sortAscending,
                     onSelectAll: widget.onSelectAll,
+                    // Make sure no decoration is set on the DataTable
+                    // from the theme, as its already wrapped in a Card.
+                    decoration: const BoxDecoration(),
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/paginated_data_table.dart:476]: This is a bit tricky, as we've discussed offline. It does seem sensible, since the decoration inherited by the DataTable within the Card is unlikely to blend with the Card's own shape and border. The API doc for PaginatedDataTable needs to make this detail clear.

File: packages/flutter/test/material/paginated_data_table_test.dart [hunk 4]
```diff
@@ -786,4 +786,34 @@ void main() {
     await tester.pumpWidget(buildTable(false));
     expect(find.byType(Checkbox), findsNothing);
   });
+
+  testWidgets('Table should not use decoration from DataTableTheme', (WidgetTester tester) async {
+    final Size originalSize = binding.renderView.size;
+    await binding.setSurfaceSize(const Size(800, 800));
+
+    Widget buildTable() => MaterialApp(
```
Comment (HansMuller) [inline packages/flutter/test/material/paginated_data_table_test.dart:794]: Prefer curly braces over `=>` for everything except one-liners that return a value.

File: packages/flutter/test/material/paginated_data_table_test.dart [hunk 5]
```diff
@@ -786,4 +786,34 @@ void main() {
     await tester.pumpWidget(buildTable(false));
     expect(find.byType(Checkbox), findsNothing);
   });
+
+  testWidgets('Table should not use decoration from DataTableTheme', (WidgetTester tester) async {
+    final Size originalSize = binding.renderView.size;
```
Comment (HansMuller) [inline packages/flutter/test/material/paginated_data_table_test.dart:791]: Is setSurfaceSize necessary for this test?

**Review by HansMuller — APPROVED** [review]
LGTM so long as the API doc makes it clear that we're ignoring the inherited DataTableTheme's decoration.

File: packages/flutter/test/material/paginated_data_table_test.dart (hunk 5 above)
Comment (perclasson) [reply packages/flutter/test/material/paginated_data_table_test.dart:791]: Yes, it's seems like the PaginatedDataTable cannot be rendered without it as it will overflow if the screen is too small. It's the same for most of the tests in this file.

File: packages/flutter/test/material/paginated_data_table_test.dart (hunk 4 above)
Comment (perclasson) [reply packages/flutter/test/material/paginated_data_table_test.dart:794]: Done

File: packages/flutter/lib/src/material/paginated_data_table.dart (hunk 3 above)
Comment (perclasson) [reply packages/flutter/lib/src/material/paginated_data_table.dart:476]: I added some docs now to the PaginatedDataTable to explain this.

File: packages/flutter/lib/src/material/paginated_data_table.dart [hunk 6]
```diff
@@ -61,6 +61,12 @@ class PaginatedDataTable extends StatefulWidget {
   ///
   /// The [rowsPerPage] and [availableRowsPerPage] must not be null (they
   /// both have defaults, though, so don't have to be specified).
+  ///
+  /// The [DataTable] is themed with [DataTableTheme]. The
+  /// [DataTableThemeData.decoration] is not applied to the [DataTable] within
+  /// a [PaginatedDataTable], as it's unlikely that it blends well with the
+  /// [Card]'s own shape and border. To modify the border or background color of
+  /// the [PaginatedDataTable], use [CardTheme].
```
Comment (guidezpl) [inline packages/flutter/lib/src/material/paginated_data_table.dart:69]:
```suggestion
  /// Themed by [DataTableTheme]. [DataTableThemeData.decoration] is ignored.
  /// To modify the border or background color of the [PaginatedDataTable], use 
  /// [CardTheme], since a [Card] wraps [PaginatedDataTable].
```

**Review by guidezpl — APPROVED** [review]
```
  _    ___ _____ __  __ 
 | |  / __|_   _|  \/  |
 | |_| (_ | | | | |\/| |
 |____\___| |_| |_|  |_|
                        
```
